static/app/components/assistant/guideAnchor.tsx

```diff
@@ -33,216 +33,180 @@ type Props = {
     pathname: string;
     query: Query;
   };
-};
-
-type State = {
-  active: boolean;
-  orgId: string | null;
-  step: number;
-  currentGuide?: Guide;
-};
-
-class BaseGuideAnchor extends Component<Props, State> {
-  state: State = {
-    active: false,
-    step: 0,
-    orgId: null,
-  };
-
-  componentDidMount() {
-    const {target} = this.props;
-    target && registerAnchor(target);
-  }
-
-  componentDidUpdate(_prevProps: Props, prevState: State) {
-    if (this.containerElement.current && !prevState.active && this.state.active) {
-      try {
-        const {top} = this.containerElement.current.getBoundingClientRect();
-        const scrollTop = window.pageYOffset;
-        const centerElement = top + scrollTop - window.innerHeight / 2;
-        window.scrollTo({top: centerElement});
-      } catch (err) {
-        Sentry.captureException(err);
-      }
+}>;
+
+function BaseGuideAnchor({
+  target,
+  onFinish,
+  children,
+  position,
+  offset,
+  containerClassName,
+  to,
+}: Props) {
+  const guideContainerRef = useRef<HTMLSpanElement>(null);
+  const {orgId, forceHide, currentGuide, currentStep} = useLegacyStore(GuideStore);
+
+  const isActive =
+    forceHide !== true && currentGuide?.steps[currentStep]?.target === (target ?? false);
+
+  // Register the anchor in the store
+  useEffect(() => {
+    if (!target) {
+      return () => {};
```

Comment:
```suggestion
      return undefined
```

Reply:
it complains that the return type is inconsistent if I do this.

Reply:
Wait really? I'm pretty sure we do it all the time when we early return from hooks - see [this](https://github.com/getsentry/sentry/blob/master/static/app/utils/profiling/hooks/useProfiles.tsx#L50) file for example, but there should be plenty other ones that do this.

Reply:
nvm you're right, I had a bare return which is apparently different 🤔